packages/cookies/src/cookies.ts

```diff
@@ -0,0 +1,41 @@
+import type { CookieSerializeOptions } from 'cookie'
+import * as cookie from 'cookie'
+
+const normalizeCookieOptions = (options: CookieSerializeOptions) => {
+  options = Object.assign({}, options)
+
+  if (options.maxAge) {
+    options.expires = new Date(Date.now() + options.maxAge * 1000)
+  }
+
+  if (options.path == null) {
+    options.path = '/'
+  }
+
+  return options
+}
+
+const serializeValue = (value: unknown) =>
+  typeof value === 'object' ? `j:${JSON.stringify(value)}` : String(value)
+
+export class Cookies extends Map<string, string> {
+  constructor(input?: string | null) {
+    const parsedInput = typeof input === 'string' ? cookie.parse(input) : {}
+    super(Object.entries(parsedInput))
+  }
+
+  set(key: string, value: unknown, options: CookieSerializeOptions = {}) {
```

Comment:
**suggestion:** given it extends a `<string, string>` map, we should expect a string value. Or relax the map type.

And if we drop the object/array support, we should check the type at runtime and throw accordingly.

Reply:
I agree about this, but that's the current implementation we have in Next.js